sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Objects;
+
+/**
+ * Represents a range of bytes within an HTTP resource.
+ * <p>
+ * The range starts at the {@link #getOffset() offset} inclusively and ends at {@link #getOffset() offset} + {@link
+ * #getLength() legnth} exclusively, or offset + length - 1.
+ * <p>
+ * If {@link #getLength() length} is unspecified, null, then the range extends to the end of the HTTP resource.
+ */
+public final class HttpRange {
```

Comment:
Should this go into core-experimental first to vet out that it works for ACS with their download APIs ?

Reply:
I can move this into experimental for now